### PR TITLE
Integrate position controller into NJointBlmcRobotDriver

### PR DESCRIPTION
**Pull request by luator**
# Description
Implement a PD position controller in the NJointBlmcRobotDriver that is
executed when passing a target position in the Action.

The controller is executed when applying an action with non-NaN target position. The resulting torque command is added to the torque given in the action.  Position control can be disabled for single joints by setting the target position for this joint to NaN.

Extend real finger demo with position controller: Implement two
functions in the demo: One implementing the position controller in
Python and sending torque commands, the other sending directly position
commands.

Further add a demo `demo_sparse_position_control.py` that shows how to only control position of some joints.

# Do not merge before
https://github.com/open-dynamic-robot-initiative/robot_interfaces/pull/6

### CODE REVIEW

File: include/blmc_robots/n_joint_blmc_robot_driver.hpp [hunk 1]
```diff
@@ -190,15 +193,73 @@ class NJointBlmcRobotDriver
...
+        {
+            // Replace NaN-values with default gains
+            applied_action.position_kp =
+                applied_action.position_kp
+                    .unaryExpr([](double x) { return std::isnan(x); })
+                    .select(default_position_kp_, desired_action.position_kp);
+            applied_action.position_kd =
+                applied_action.position_kd
+                    .unaryExpr([](double x) { return std::isnan(x); })
+                    .select(default_position_kd_, desired_action.position_kd);
```
Comment (mwuethri) [inline include/blmc_robots/n_joint_blmc_robot_driver.hpp]: very elegant :)

File: include/blmc_robots/n_joint_blmc_robot_driver.hpp [hunk 2]
```diff
@@ -190,15 +193,73 @@ class NJointBlmcRobotDriver
...
+
+            Vector position_error = applied_action.position - observation.angle;
```
Comment (mwuethri) [inline include/blmc_robots/n_joint_blmc_robot_driver.hpp]: i guess we should now also change it from angle to position in the Observation class.

File: include/blmc_robots/n_joint_blmc_robot_driver.hpp [hunk 3]
```diff
@@ -190,15 +193,73 @@ class NJointBlmcRobotDriver
...
+                applied_action.position_kp.cwiseProduct(position_error) +
+                applied_action.position_kd.cwiseProduct(observation.velocity);
+
+            // position_control_torque contains NaN for joints where target
+            // position is set to NaN!  Filter those out and set the torque to
+            // zero instead.
+            position_control_torque =
+                position_control_torque
+                    .unaryExpr([](double x) { return std::isnan(x); })
+                    .select(0, position_control_torque);
```
Comment (mwuethri) [inline include/blmc_robots/n_joint_blmc_robot_driver.hpp]: i think it is ok here, but we once noticed that C++ operations involving nans are much slower than operations involving just regular doubles, so generally we should avoid that (i.e. detect nans before operations are done on them).

File: include/blmc_robots/n_joint_blmc_robot_driver.hpp (hunk 2 above)
Comment (luator) [reply include/blmc_robots/n_joint_blmc_robot_driver.hpp]: Yes, I will do this next

File: include/blmc_robots/n_joint_blmc_robot_driver.hpp (hunk 3 above)
Comment (luator) [reply include/blmc_robots/n_joint_blmc_robot_driver.hpp]: Good to know, I wasn't aware of that.

**Pull request merged**